Source/Blazorise/Components/FocusTrap/FocusTrap.razor.cs

```diff
@@ -0,0 +1,84 @@
+﻿#region Using directives
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Web;
+#endregion
+
+namespace Blazorise
+{
+    /// <summary>
+    /// The focus trap component allows to restrict TAB key navigation inside the component.
+    /// </summary>
+    public partial class FocusTrap : BaseComponent
+    {
+        #region Members
+
+        private ElementReference containerRef;
+
+        private ElementReference startFirstRef;
+
+        private ElementReference startSecondRef;
+
+        private ElementReference endFirstRef;
+
+        private ElementReference endSecondRef;
+
+        private bool _shiftPressed;
+
+        #endregion
+
+        #region Methods
+
+        /// <inheritdoc/>
+        protected override async Task OnAfterRenderAsync( bool firstRender )
+        {
+            if ( firstRender )
+            {
+                await startFirstRef.FocusAsync();
+            }
+        }
+
+        internal async Task SetFocus()
+            => await startFirstRef.FocusAsync();
+
+        private async Task FocusStartAsync( FocusEventArgs args )
+        {
+            if ( !_shiftPressed )
+            {
+                await startFirstRef.FocusAsync();
+            }
+        }
+
+        private async Task FocusEndAsync( FocusEventArgs args )
+        {
+            if ( _shiftPressed )
+            {
+                await endFirstRef.FocusAsync();
+            }
+        }
+
+        private void HandleKeyPresses( KeyboardEventArgs args )
+        {
+            if ( args.Key == "Tab" )
+            {
+                _shiftPressed = args.ShiftKey;
```

Comment:
No biggy, but maybe rename to _shiftTabPressed? Which is what it represents? I was reading the code and I was like, wth is this _shiftPressed thing until I got to this method.

Reply:
Haha yeah you're right. I thought the same when I seen it but then forgot about it.